### PR TITLE
feat(storage-browser): add useView

### PR DESCRIPTION
**Pull request by calebpollman**
#### Description of changes
Add `useView`, split off from the larger `View` context work
- add `createUseView` util
- add missing action configs
- scaffold _LocationActionView_ directory structure in preparation of moving to isolated views (e.g. `StorageBrowser.UploadView`)
- add new types (some of which are unrelated to this PR but difficult to decouple from the `View` context work)
- move `useDeleteActionView` hook and tests, minor API updates
- will add unit tests and improve robustness in follow up

#### Checklist

- [ ] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [ ] PR description included
- [ ] `yarn test` passes and tests are updated/added
- [ ] PR title and commit messages follow [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) syntax
- [ ] _If this change should result in a version bump_, [changeset added](https://github.com/changesets/changesets/blob/main/docs/adding-a-changeset.md) (This can be done after creating the PR.) This does not apply to changes made to `docs`, `e2e`, `examples`, or other private packages.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by dbanksdesign — APPROVED** [review]

File: packages/react-storage/src/components/StorageBrowser/views/createUseView.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+): viewName is keyof DefaultUseViews =>
+  Object.keys(DEFAULT_USE_VIEWS).some((key) => key === viewName);
+
+export const createUseView: CreateUseView = (configs) => {
+  const hooks: Record<string, UseView> = Object.values(configs).reduce(
+    (out, { componentName }) => ({
+      ...out,
+      [componentName.slice(0, -4)]: isDefaultUseViewName(componentName)
+        ? DEFAULT_USE_VIEWS[componentName]
+        : useActionView,
```
Comment (reesscot) [inline packages/react-storage/src/components/StorageBrowser/views/createUseView.ts:53]: Is this for custom actions?

Comment (calebpollman) [reply packages/react-storage/src/components/StorageBrowser/views/createUseView.ts:53]: Yep, had more happening with it on the original branch but didn't want to fully tear it out

**Review by reesscot — DISMISSED** [review]

File: packages/react-storage/src/components/StorageBrowser/views/LocationActionView/useActionView.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import { ActionViewState } from './types';
+
+export function useActionView(): ActionViewState {
```
[the remark this answers is not in the document]
Comment (calebpollman) [reply packages/react-storage/src/components/StorageBrowser/views/LocationActionView/useActionView.ts:3]: This is the hook for custom actions

**Review by reesscot — APPROVED** [review]

File: examples/next/pages/ui/components/storage/storage-browser/managed-auth/routed/[locations]/index.page.tsx [hunk 3]
```diff
@@ -4,11 +4,34 @@ import { useRouter } from 'next/router';
 import { Flex } from '@aws-amplify/ui-react';
 
 import { SignOutButton } from '../components';
-import { StorageBrowser } from '../StorageBrowser';
+import { StorageBrowser, useView } from '../StorageBrowser';
 
 import '@aws-amplify/ui-react-storage/storage-browser-styles.css';
 import '@aws-amplify/ui-react-storage/styles.css';
 
+function TemporaryLogs() {
```
Comment (AllanZhengYP) [inline examples/next/pages/ui/components/storage/storage-browser/managed-auth/routed/[locations]/index.page.tsx:12]: Is this intentional?

**Review by dbanksdesign — APPROVED** [review]

**Review by AllanZhengYP — APPROVED** [review]

**Pull request merged**